File: recursion/capitalize-words.js
let capitalizeWords = (arr) => {

  if (arr.length === 0) return;

  // should return an array with capitalized words
  let wordsArray = []

  const capitalizeHelper = (arr) => {
    if (arr.length === 0) return;
    const firstWord = arr[0].split('')
    const capitalizedWord = firstWord.map((l) => l.toUpperCase())
    wordsArray.push(capitalizedWord.join(''))
    const filteredArr = arr.filter((word) => word !== arr[0])
    capitalizeHelper(filteredArr);
  }

  capitalizeHelper(arr)

  return wordsArray;
}

console.log(capitalizeWords(["hello", "bye", "adios"]))

File: recursion/capitalize-words.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main()
    {
        List<string> words = new List<string> { "hello", "bye", "adios" };
        var capitalizedWords = CapitalizeWords(words);
        Console.WriteLine(String.Join(", ", capitalizedWords));
    }

    static List<string> CapitalizeWords(List<string> arr)
    {
        if (arr.Count == 0) return new List<string>();

        List<string> wordsArray = new List<string>();

        void CapitalizeHelper(List<string> array)
        {
            if (array.Count == 0) return;
            string firstWord = array[0];
            string capitalizedWord = firstWord.ToUpper();
            wordsArray.Add(capitalizedWord);

            List<string> filteredArr = array.Where(word => word != array[0]).ToList();
            CapitalizeHelper(filteredArr);
        }

        CapitalizeHelper(arr);
        return wordsArray;
    }
}
